src/constants.ts

```diff
@@ -724,7 +723,7 @@ declare const BOOSTS: {
             damage: 0.3;
         };
     };
-};
+} & Record<BodyPartConstant, Record<MineralBoostConstant, Record<BoostModifier, number>>>;
```

Comment:
```suggestion
} & Record<BodyPartConstant, Record<MineralBoostConstant, Record<BoostModifier, number>>>;
```
I think this change ruins some indexing into `BOOSTS` that people may expect to do.
I would argue that there are better solutions to that - I don't even think this Record type should be there, nor the `BoostModifier` type.
Normal iterations result in a key of type `string`, thus you can only use it on objects with an index signature `Record<string, T>`.

I think it is a bad idea to just add that to all objects though...

Reply:
I don't know what's the fix then. Right now, the code I have fails unless I overlay that over `BOOSTS`. And to be fair the initial issue is about `creep.body[0].boost` being wrong. That's just the part I brought from OM since I think it acts as documentation over what the keys actually are.

Also, I'm using my own wrappers over `Object.keys/values/entries` to make them return the correct key/value/tuple type instead of `string`, which maybe that's just a me thing.